mmv1/third_party/terraform/tpgresource/utils_test.go

```diff
@@ -37,6 +37,145 @@ var fictionalSchema = map[string]*schema.Schema{
+			want:       []string{"one", "three", "two"},
+		},
+		"config order is preserved": {
+			configData: []string{"foo", "two", "bar", "baz"},
+			apiData:    []string{"one", "two", "three", "bar"},
+			want:       []string{"two", "bar", "one", "three"},
+		},
+		"config data and api data overlap completely": {
+			configData: []string{"foo", "bar", "baz", "one", "two", "three"},
+			apiData:    []string{"baz", "two", "one", "bar", "three", "foo"},
+			want:       []string{"foo", "bar", "baz", "one", "two", "three"},
+		},
+		"config data contains duplicates": {
+			configData: []string{"one", "one"},
+			apiData:    []string{},
+			wantError:  true,
+		},
+		"api data contains duplicates": {
+			configData: []string{},
+			apiData:    []string{"one", "one"},
+			wantError:  true,
+		},
+	}
+
+	for tn, tc := range cases {
+		tc := tc
+		t.Run(fmt.Sprintf("strings/%s", tn), func(t *testing.T) {
+			t.Parallel()
+			sorted, err := SortStringsByConfigOrder(tc.configData, tc.apiData)
+			if err != nil {
+				if !tc.wantError {
+					t.Fatalf("Unexpected error: %s", err)
+				}
+			} else if tc.wantError {
+				t.Fatalf("Wanted error, got none")
+			}
+			if !tc.wantError && sorted != tc.want {
+				t.Fatalf("sorted result is incorrect. want %v, got %v", sorted, tc.want)
+			}
+		})
+
+		t.Run(fmt.Sprintf("maps/%s", tn), func(t *testing.T) {
+			t.Parallel()
+			configData := []map[string]interface{}{}
+			for _, item := range tc.configData {
+				configData = append(configData, map[string]interface{}{
+					"value": item,
+				})
+			}
+			apiData := []map[string]interface{}{}
+			for _, item := range tc.apiData {
+				apiData = append(apiData, map[string]interface{}{
+					"value": item,
+				})
+			}
+			sorted, err := SortMapsByConfigOrder(configData, apiData, "value")
+			if err != nil {
+				if !tc.wantError {
+					t.Fatalf("Unexpected error: %s", err)
+				}
+			} else if tc.wantError {
+				t.Fatalf("Wanted error, got none")
+			}
+			if !tc.wantError && sorted != tc.want {
+				t.Fatalf("sorted result is incorrect. want %v, got %v", sorted, tc.want)
+			}
+		})
+	}
+}
+
+func TestSortMapsByConfigOrder(t *testing.T) {
+	// most cases are covered by TestSortByConfigOrder; this covers map-specific cases.
+	cases := struct {
+		configData, apiData []map[string]interface{}
+		idKey               string
+		wantError           bool
+		want                []map[string]interface{}
+	}{
+		"config data is malformed": {
+			configData: map[string]interface{}{
+				"foo": "one",
+			},
+			apiData:   map[string]interface{}{},
+			idKey:     "bar",
+			wantError: true,
+		},
+		"api data is malformed": {
+			configData: map[string]interface{}{},
+			apiData: map[string]interface{}{
+				"foo": "one",
+			},
+			idKey:     "bar",
+			wantError: true,
+		},
+	}
+
+	for tn, tc := range cases {
+		tc := tc
+		t.Run(tn, func(t *testing.T) {
+			t.Parallel()
+			sorted, err := SortMapsByConfigOrder(tc.configData, tc.apiData, tc.idKey)
```

Comment:
```suggestion
			sorted, err := tpgresource.SortMapsByConfigOrder(tc.configData, tc.apiData, tc.idKey)
```

I think the tests all need to be updated in a similar way to access the new functions

Reply:
Fixed in 0e3ccd403